core/src/domain/channel/entities.rs

```diff
@@ -0,0 +1,169 @@
+use chrono::{DateTime, Utc};
+use thiserror::Error;
+use uuid::Uuid;
+
+use crate::domain::server::entities::ServerId;
+
+pub const MAX_CHANNEL_NAME_SIZE: usize = 30;
+
+pub const MIN_CHANNEL_NAME_SIZE: usize = 2;
+
+#[derive(Error, Debug, Clone)]
+pub enum ChannelError {
+    #[error(
+        "Channel name is too long. It should not be longer than {}",
+        MAX_CHANNEL_NAME_SIZE
+    )]
+    ChannelNameTooLong,
+
+    #[error(
+        "Channel name is too short. It should not be  than {}",
+        MIN_CHANNEL_NAME_SIZE
+    )]
+    ChannelNameTooShort,
+
+    #[error("Channel type is incorrect")]
+    WrongChannelType,
+
+    #[error("The payload to update the channel is empty")]
+    EmptyUpdatePayload,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Default)]
+pub struct ChannelId(pub Uuid);
+
+/// The string is the value of the name
+/// The Option<bool> represetant if the state of the validation
+/// If the validation is not already
+#[derive(Clone, Debug)]
+pub struct ChannelName(String, Option<bool>);
+
+impl ChannelName {
+    pub fn new(value: String) -> Self {
+        let formated_value = value.trim().to_string();
+        Self(formated_value, None)
+    }
+
+    fn is_too_long(&self) -> bool {
+        self.0.len() > MAX_CHANNEL_NAME_SIZE
+    }
+
+    fn is_too_short(&self) -> bool {
+        self.0.len() < MIN_CHANNEL_NAME_SIZE
+    }
+
+    pub fn check(&mut self) -> Result<(), ChannelError> {
+        if self.is_too_long() {
+            self.1 = Some(false);
+            return Err(ChannelError::ChannelNameTooLong);
+        }
+        if self.is_too_short() {
+            self.1 = Some(false);
+            return Err(ChannelError::ChannelNameTooShort);
+        }
+        Ok(())
+    }
+
+    pub fn is_valid(&mut self) -> bool {
+        if let Some(is_valid) = self.1 {
+            is_valid
+        } else {
+            self.check().is_err()
+        }
+    }
+
+    pub fn value(&mut self) -> Result<String, ChannelError> {
```

Comment:
Is the clone really usefull ?

```suggestion
    pub fn value(&mut self) -> Result<String, ChannelError> {
        self.check()?;
        Ok(self.0.clone())
    }
```